src/axom/sidre/datacollection/SidreDataCollection.hpp

```diff
@@ -0,0 +1,528 @@
+              │    │    │    └─• boundary_topology = "boundary"
+              │    │    └── boundary
+              │    │         ├─• type = "unstructured"
+              │    │         ├── elements
+              │    │         │    ├─• shape = "points"|"lines"|...
+              │    │         │    └─• connectivity = <vert-idx-array>
+              │    │         └─• coordset = "coords"
+              │    └── fields
+              │         ├── mesh_material_attribute
+              │         │    ├─• association = "element"
+              │         │    ├─• topology = "mesh"
+              │         │    └─• values = <attr-array>
+              │         ├── boundary_material_attribute
+              │         │    ├─• association = "element"
+              │         │    ├─• topology = "boundary"
+              │         │    └─• values = <attr-array>
+              │         ├── grid-function-1   (name can include path)
+              │         │    ├─• basis = "<fe-coll-name>"
+              │         │    ├─• topology = "mesh"
+              │         │    └─• values = <ext-double-array>/<named-buffer> (vdim == 1)
+              │         ├── grid-function-2   (name can include path)
+              │         │    ├─• basis = "<fe-coll-name>"
+              │         │    ├─• topology = "mesh"
+              │         │    └── values   (vdim > 1)
+              │         │         ├─• x0 = view into <ext-double-array>/<named-buffer>
+              │         │         ├─• x1 = view into <ext-double-array>/<named-buffer>
+              │         │         └─• x2 = view into <ext-double-array>/<named-buffer>
+              │         ├── ...
+              │        ...
+              └── named_buffers                (named_buffers group)
+                   ├─• vertex_coords = <double-array>
+                   ├─• grid-function-1 = <double-array>
+                   ├─• grid-function-2 = <double-array>
+                  ...
+
+    @note blueprint_index is used both in serial and in parallel. In parallel,
+    only rank 0 will add entries to the blueprint index.
+
+    @note QuadratureFunction%s (q-fields) are not supported.
+
+    @note SidreDataCollection does not manage the FiniteElementSpace%s and
+    FiniteElementCollection%s associated with registered GridFunction%s.
+    Therefore, field registration is left to the user of SidreDataCollection and
+    there are no methods that automatically register GridFunction%s using just
+    the content of the Sidre DataStore. Such capabilities can be implemented in
+    a derived class, adding any desired object management routines.
+
+    @warning This class is still _experimental_, meaning that in future
+    releases, it may not be backward compatible, and the output files generated
+    by the current version may become unreadable.
+*/
+class SidreDataCollection : public mfem::DataCollection
+{
+public:
+   typedef mfem::NamedFieldsMap< mfem::Array<int> > AttributeFieldMap;
```

Comment:
minor style: in axom we tend to prefer the _alias_ syntax instead of the `typedef`:
```
using AtributeFieldMap = mfem::NameFieldMap< mfem::Array<int> >;
```

Reply:
Good catch, I went ahead and made some other C++11 updates as well (e.g. `NULL` to `nullptr`.)